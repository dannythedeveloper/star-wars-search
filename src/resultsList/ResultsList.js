import React from 'react';
import Item from '../item/Item';

export default class ResultsList extends React.Component {
    getResults = () => {
        if(this.props.loading) {
            return (
                <h2>Loading your results...</h2>
            )
        } else if(this.props.blankResults && this.props.loading === false) {
            return (
                <h2>There are no results to display. Check your search inputs and try again.</h2>
            );
        }
        let searchList = this.props.results.map((result, index) => {
            return (<Item
                        key={index}
                        name={result.name}
                        title={result.title}    
                    />
            )
        })
        return searchList;
    }

    render() {
        return (
            <div>
                {this.getResults()}
            </div>
        )
    }
}