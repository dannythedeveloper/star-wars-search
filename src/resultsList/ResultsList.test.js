import React from 'react';
import ReactDOM from 'react-dom';
import ResultsList from './ResultsList';
import renderer from 'react-test-renderer';

describe('ResultsList Component', () => {
    it('render without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<ResultsList 
            results={ [
                {
                  birth_year: "19 BBY",
                  eye_color: "Blue",
                  films: ["https://swapi.co/api/films/1/"],
                  gender: "Male",
                  hair_color: "Blond",
                  height: "172",
                  homeworld: "https://swapi.co/api/planets/1/",
                  mass: "77",
                  name: "Luke Skywalker",
                  skin_color: "Fair",
                  created: "2014-12-09T13:50:51.644000Z",
                  edited: "2014-12-10T13:52:43.172000Z",
                  species: ["https://swapi.co/api/species/1/"],
                  starships: ["https://swapi.co/api/starships/12/"],
                  url: "https://swapi.co/api/people/1/",
                  vehicles: ["https://swapi.co/api/vehicles/14/"],
                },
              ] }
        />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI', () => {
        const tree = renderer.create(<ResultsList
            results={ [
                {
                  birth_year: "19 BBY",
                  eye_color: "Blue",
                  films: ["https://swapi.co/api/films/1/"],
                  gender: "Male",
                  hair_color: "Blond",
                  height: "172",
                  homeworld: "https://swapi.co/api/planets/1/",
                  mass: "77",
                  name: "Luke Skywalker",
                  skin_color: "Fair",
                  created: "2014-12-09T13:50:51.644000Z",
                  edited: "2014-12-10T13:52:43.172000Z",
                  species: ["https://swapi.co/api/species/1/"],
                  starships: ["https://swapi.co/api/starships/12/"],
                  url: "https://swapi.co/api/people/1/",
                  vehicles: ["https://swapi.co/api/vehicles/14/"],
                },
              ] }    
        />).toJSON();
        expect(tree).toMatchSnapshot();
    })
});