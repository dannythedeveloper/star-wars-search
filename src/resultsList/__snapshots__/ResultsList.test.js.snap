// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsList Component renders the UI 1`] = `
<div>
  <div
    className="item"
  >
    <h2>
      Luke Skywalker
    </h2>
  </div>
</div>
`;
